drivers/serial/uart_ns16550.c

```diff
@@ -272,7 +272,7 @@ struct uart_ns16550_device_config {
 /** Device data structure */
 struct uart_ns16550_dev_data_t {
 #ifdef UART_NS16550_PCIE_ENABLED
-	struct uart_device_config devconf;
+	uint64_t pcimem;
```

Comment:
Why not uintptr_t directly? (can we get 64bits BAR in 32 bits?)

Actually, uart_device_config base attribute could be changed to uintprt_t but that's out of scope here.

Reply:
Well... the 64 bit field is a hardware feature, not something about OS code generation.  That second word is there regardless of whether the CPU is running in long mode or not.  But you're right, since we can't use the high bits, there's no particular reason to store them in the driver in 32 bit mode.